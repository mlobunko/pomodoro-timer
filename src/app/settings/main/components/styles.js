import styled from "styled-components";
import ImageBgTomato from "../images/bg-tomato.svg";

export const StyledButtons = styled.div`
  margin-top: 1.2rem;
  margin-bottom: 0.5rem;
  flex-shrink: 0;
  width: 31rem;
  height: 4.1rem;
  display: flex;
  justify-content: center;

  @media (min-width: 360px) {
    width: 35rem;
  }

  @media (min-width: 600px) {
    width: 42rem;
  }
`;
StyledButtons.displayName = "StyledButtons";

export const StyledCenterOfPage = styled.div`
  height: 65%;
  width: 31rem;
  flex: 1 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  font-size: 3.2rem;

  @media (min-width: 360px) {
    width: 35rem;
  }
  @media (min-width: 600px) {
    width: 42rem;
    height: 80%;
  }
`;
StyledCenterOfPage.displayName = "StyledCenterOfPage";

export const StyledHeader = styled.h1`
  margin: 0;
  flex-shrink: 0;
  text-align: center;
`;
StyledHeader.displayName = "StyledHeader";

export const StyledListOfSettings = styled.div`
  display: grid;
  grid-template-columns: 40% 60%;
  grid-template-rows: 5rem 5rem 5rem 5rem;
  grid-gap: 1rem;
  justify-items: center;
  align-items: center;
`;
StyledListOfSettings.displayName = "StyledListOfSettings";

export const StyledMainComponent = styled.div`
  background: url(${ImageBgTomato}) center;
  background-size: cover;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  align-items: center;
  font-size: 1.6rem;
  font-family: CarterOne;
`;
StyledMainComponent.displayName = "StyledMainComponent";
