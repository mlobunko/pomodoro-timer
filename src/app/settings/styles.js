import styled from "styled-components";

export const StyledInputs = styled.div`
  display: flex;
  justify-content: "space-evenly";
`;
StyledInputs.displayName = "StyledInputs";

export const StyledInput = styled.div`
  width: 4.5rem;
`;
StyledInput.displayName = "StyledInput";
