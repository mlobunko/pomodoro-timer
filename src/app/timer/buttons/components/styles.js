import styled from "styled-components";

export const StyledButtonsComponent = styled.div`
  margin-top: 1.2rem;
  margin-bottom: 0.5rem;
  flex-shrink: 0;
  width: 31rem;
  height: 4.1rem;
  display: flex;
  justify-content: space-between;

  @media (min-width: 360px) {
    width: 35rem;
  }

  @media (min-width: 600px) {
    width: 42rem;
  }
`;
StyledButtonsComponent.displayName = "StyledButtonsComponent";
