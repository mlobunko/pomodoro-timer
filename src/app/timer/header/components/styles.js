import styled from "styled-components";
import ImageTomatoCool from "shared/images/tomato-cool.svg";

export const StyledHeader = styled.h1`
  margin: 0;
  flex-shrink: 0;
  text-align: center;
  font-family: CarterOne;
`;
StyledHeader.displayName = "StyledHeader";

export const StyledInfo = styled.div`
  display: inline-block;
  margin-left: 1rem;
  font-size: 2.5rem;
`;
StyledInfo.displayName = "StyledInfo";

export const StyledTomatoIcon = styled.div`
  display: inline-block;
  background: url(${ImageTomatoCool}) no-repeat center center;
  margin-left: 0.2rem;
  background-size: cover;
  width: 4rem;
  height: 4.2rem;
`;
StyledTomatoIcon.displayName = "StyledTomatoIcon";
