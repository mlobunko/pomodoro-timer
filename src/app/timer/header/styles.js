import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styled from "styled-components";

export const StyledFontAwesomeIcon = styled(FontAwesomeIcon)`
  color: rgb(61, 153, 112);
`;
StyledFontAwesomeIcon.displayName = "StyledFontAwesomeIcon";
