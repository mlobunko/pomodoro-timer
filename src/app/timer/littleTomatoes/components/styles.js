import styled, { keyframes } from "styled-components";

import ImageTomato from "shared/images/tomato.svg";

const styleTomato = `
  background: url(${ImageTomato}) no-repeat center;
  width: 70%;
`;

const blinking = keyframes`
  0% {
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
`;

export const StyledAnimatedLittleTomato = styled.div`
  ${styleTomato};
  animation: ${blinking} 3s linear infinite;
`;
StyledAnimatedLittleTomato.displayName = "StyledAnimatedLittleTomato";

export const StyledLittleTomato = styled.div`
  ${styleTomato};
`;
StyledLittleTomato.displayName = "StyledLittleTomato";

export const StyledLittleTomatoesComponent = styled.div`
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
  grid-template-rows: 7.5rem;
  justify-items: center;
`;
StyledLittleTomatoesComponent.displayName = "StyledLittleTomatoesComponent";
