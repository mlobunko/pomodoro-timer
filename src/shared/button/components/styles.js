import styled from "styled-components";
import PropTypes from "prop-types";

export const StyledButton = styled.button`
  width: ${props => {
    if (props.round) return "4rem";
    if (props.side) return "7.5rem";
    else return "9.5rem";
  }};
  height: ${props => props.round && "4rem"};
  line-height: ${props => props.round && "4rem"};
  padding: ${props => (props.round ? "" : "0.6rem 1.6rem")};

  border: ${props => (props.round ? "1px solid #3d9970" : "2px solid #3d9970")};
  border-radius: ${props => (props.round ? "50%" : "1.2rem")};

  font-size: ${props => (props.round ? "1.6rem" : "2.2rem")};

  text-align: center;
  vertical-align: ${props => props.round && "center"};

  color: white;
  background: #3d9970;

  &:hover,
  &:active {
    color: #3d9970;
    background: white;
  }
`;

StyledButton.displayName = "StyledButton";

StyledButton.propTypes = {
  round: PropTypes.bool,
  side: PropTypes.bool
};
