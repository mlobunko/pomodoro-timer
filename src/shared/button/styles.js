import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styled from "styled-components";

export const StyledFontAwesomeIcon = styled(FontAwesomeIcon)``;

StyledFontAwesomeIcon.displayName = "StyledFontAwesomeIcon";
