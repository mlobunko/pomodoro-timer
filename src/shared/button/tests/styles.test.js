import React from "react";
import { shallow } from "enzyme";
import "enzyme-to-json";

import { StyledFontAwesomeIcon } from "../styles";

describe("StyledFontAwesomeIcon", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<StyledFontAwesomeIcon />);
    expect(wrapper).toMatchSnapshot();
  });
});
