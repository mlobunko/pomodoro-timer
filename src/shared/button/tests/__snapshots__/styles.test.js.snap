// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledFontAwesomeIcon should render correctly 1`] = `
<FontAwesomeIcon
  border={false}
  className="sc-bdVaJa bDWFJH"
  fixedWidth={false}
  flip={null}
  icon={null}
  inverse={false}
  listItem={false}
  mask={null}
  pull={null}
  pulse={false}
  rotation={null}
  size={null}
  spin={false}
  symbol={false}
  transform={null}
/>
`;
